### PR TITLE
Add themes for custom HTML infowindow migration with non-light style

### PR DESCRIPTION
**Pull request by javitonino**
Fix for https://github.com/CartoDB/cartodb/issues/11380

Adds legacy compatibility for old infowindow styles. With this, migration is much better than before, still not perfect. If you migrate the `.carto` example, you will see that the text just before the header is occluded by the header itself. This is a side-effect on how the header is implemented in the migrated version (using negative margins).

The aim was not to be perfect, but just display something reasonable. Feel free to comment on ideas to improve it, this is what I found to be a good trade-off between looks and complexity.

CR-note: I ignored the nested too deep complaints, as trying to unnest it resulted in a more chaotic sccs IMHO. If you think they are important, leave a comment and we can try to unnest a level.

screenshot_20170203_135429 screenshot_20170203_134934

## Acceptance
You can use Builder branch `11380-custom_infowindow_migration_style` which has the updated shrinkwrap.

- [x] Import the `.carto` in the issue (https://github.com/CartoDB/cartodb/issues/11380#issuecomment-277212319) and see that the end result is reasonable (there are some screenshots in the linked comment)
- [x] Create some custom HTMLs infowindows in editor and migrate to Builder. You can also test with the following HTMLs (for each theme). Use a map over this dataset: http://spatialkeydocs.s3.amazonaws.com/FL_insurance_sample.csv.zip

Dark
```
<div class="cartodb-popup dark v2">
  <a href="#close" class="cartodb-popup-close-button close">x</a>
  <div class="cartodb-popup-content-wrapper">
    <div class="cartodb-popup-content">
      <h4>construction</h4>
      <p>{{construction}}</p>
      <h4>county</h4>
      <p>{{county}}</p>
      <h4>eq_site_deductible</h4>
      <p>{{eq_site_deductible}}</p>
    </div>
  </div>
  <div class="cartodb-popup-tip-container"></div>
</div>
```

Light
```
<div class="cartodb-popup v2">
  <a href="#close" class="cartodb-popup-close-button close">x</a>
  <div class="cartodb-popup-content-wrapper">
    <div class="cartodb-popup-content">
      <h4>construction</h4>
      <p>{{construction}}</p>
      <h4>county</h4>
      <p>{{county}}</p>
      <h4>eq_site_deductible</h4>
      <p>{{eq_site_deductible}}</p>
    </div>
  </div>
  <div class="cartodb-popup-tip-container"></div>
</div>
```

Blue
```
<div class="cartodb-popup header blue v2">
  <a href="#close" class="cartodb-popup-close-button close">x</a>
  <div class="cartodb-popup-header">
    <h4>construction</h4>
    <h1>{{construction}}</h1>
    <span class="separator"></span>
  </div>
  <div class="cartodb-popup-content-wrapper">
    <div class="cartodb-popup-content">
      <h4>county</h4>
      <p>{{county}}</p>
      <h4>eq_site_deductible</h4>
      <p>{{eq_site_deductible}}</p>
    </div>
  </div>
  <div class="cartodb-popup-tip-container">
  </div>
</div>
```

Green
```
<div class="cartodb-popup header green v2">
  <a href="#close" class="cartodb-popup-close-button close">x</a>
  <div class="cartodb-popup-header">
    <h4>construction</h4>
    <h1>{{construction}}</h1>
    <span class="separator"></span>
  </div>
  <div class="cartodb-popup-content-wrapper">
    <div class="cartodb-popup-content">
      <h4>county</h4>
      <p>{{county}}</p>
      <h4>eq_site_deductible</h4>
      <p>{{eq_site_deductible}}</p>
    </div>
  </div>
  <div class="cartodb-popup-tip-container">
  </div>
</div>
```

Yellow
```
<div class="cartodb-popup header yellow v2">
  <a href="#close" class="cartodb-popup-close-button close">x</a>
  <div class="cartodb-popup-header">
    <h4>construction</h4>
    <h1>{{construction}}</h1>
    <span class="separator"></span>
  </div>
  <div class="cartodb-popup-content-wrapper">
    <div class="cartodb-popup-content">
      <h4>county</h4>
      <p>{{county}}</p>
      <h4>eq_site_deductible</h4>
      <p>{{eq_site_deductible}}</p>
    </div>
  </div>
  <div class="cartodb-popup-tip-container">
  </div>
</div>
```

Orange
```
<div class="cartodb-popup header orange v2">
  <a href="#close" class="cartodb-popup-close-button close">x</a>
  <div class="cartodb-popup-header">
    <h4>construction</h4>
    <h1>{{construction}}</h1>
    <span class="separator"></span>
  </div>
  <div class="cartodb-popup-content-wrapper">
    <div class="cartodb-popup-content">
      <h4>county</h4>
      <p>{{county}}</p>
      <h4>eq_site_deductible</h4>
      <p>{{eq_site_deductible}}</p>
    </div>
  </div>
  <div class="cartodb-popup-tip-container">
  </div>
</div>
```

### CODE REVIEW

**Comment by xavijam** [comment]
Could you check this PR @piensaenpixel please?

**Comment by matallo** [comment]
did the proposed changes myself, another pair of 👀 won't hurt

**Comment by xavijam** [comment]
CR sounds legit, do you want me to test it?

**Comment by matallo** [comment]
that's part of https://github.com/CartoDB/cartodb/issues/11380 (I already did some test myself) go!

**Comment by xavijam** [comment]
Dale manolo!

**Comment by javitonino** [comment]
Updated acceptance branch (for shrinkwrap), ready for anyone to take it.

**Comment by javierarce** [comment]
It looks pretty good!

I only found a minor issue with the top padding of the content in the popups/infowindows with headers:

screen shot 2017-02-08 at 15 20 25

I'd add a padding top of 16px to `.cartodb-popup-content` in order to have this:

screen shot 2017-02-08 at 15 20 21

I would also reduce the size of the header tags that are inside of the popup header, so they are smaller and resemble the ones we are currently offering:

screen shot 2017-02-08 at 15 25 07

Since the fixes are quite small and straightforward, I would move it back to the development, but I open this for discussion.

**Comment by javitonino** [comment]
Agree with 1 and 2. Disagree with 3, I think we should keep the bigger font like in the original popup.

screenshot_20170208_170505

I'll also uppercase the font, so it looks like this:
screenshot_20170208_170752

What do you think @javierarce?

**Comment by javierarce** [comment]
I like it! 

Regarding #3: I don't have a strong opinion and I think that respecting the original size is a valid point, so fine by me :)

File: themes/scss/infowindow/cartodb-infowindow-legacy.scss [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+      color: #FFF;
+      font: bold 21px "Helvetica Neue", "Helvetica", Arial;
+      line-height: 23px;
+      text-shadow: 0 1px rgba(0, 0, 0, 0.5);
+      word-wrap: break-word;
+      text-transform: uppercase;
+    }
+
+    h4 {
+      text-transform: uppercase;
```
Comment (xavijam) [inline themes/scss/infowindow/cartodb-infowindow-legacy.scss]: I ❤️  you Hound

Comment (javitonino) [inline themes/scss/infowindow/cartodb-infowindow-legacy.scss]: You troll :D

**Pull request merged**